PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppRunState.java

```diff
@@ -177,8 +184,12 @@ private WebAppBase getDeployTargetByConfiguration(@NotNull RunProcessHandler pro
             throw new Exception(NO_WEB_APP);
         }
 
-        if (webAppSettingModel.isDeployToSlot()) {
-            return webApp.deploymentSlots().getByName(webAppSettingModel.getSlotName());
+        if (isDeployToSlot()) {
+            if (webAppSettingModel.getSlotName() == Constants.CREATE_NEW_SLOT) {
```

Comment:
It would be better to add a flag in `webAppSettingModel`, and use the flag here to determine what to do next.

Reply:
Will do it together with refactoring the logic of reset from config and render ui elements by user behaviors.